app/js/tests/reducers/test_form_content.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+var test = require( 'tape' ),
+	deepFreeze = require( 'deep-freeze' ),
+	formContent = require( '../../lib/reducers/form_content' );
+
+test( 'SELECT_AMOUNT sets amount and isCustomAmount', function ( t ) {
+	var stateBefore = { amount: 99, isCustomAmount: true },
+		expectedState = { amount: 5, isCustomAmount: false };
+
+	deepFreeze( stateBefore );
+	t.deepEqual( formContent( stateBefore, { type: 'SELECT_AMOUNT', payload: { amount: 5 } } ), expectedState );
+	t.end();
+
+} );
+
+test( 'SELECT_AMOUNT keeps amount if selected amount is null', function ( t ) {
+	var stateBefore = { amount: 99, isCustomAmount: true },
+		expectedState = { amount: 99, isCustomAmount: false };
```

Comment:
When the user enters the input field, the selected amount is deleted. By keeping the amount, it can be restored when the user leaves the input field without entering something.